lib/types/signer.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Kadena.Types.Signer do
+  @moduledoc """
+  `Signer` struct definition.
+  """
+
+  alias Kadena.Types.{Base16String, CapsList, OptionalCapsList}
+
+  @behaviour Kadena.Types.Spec
+
+  @valid_schemes [:ed25519, nil]
```

Comment:
@CristhianRodriguezMolina @santiagoboterokommit I'm not so sure that `nil` is a valid "scheme". IMO, you should remove it from this list and use a pattern matching in the validation function instead.

Reply:
it is to make scheme optional, the scheme could not be given, so the default value will be nil.
and it is not necessary to make an OptionalScheme because here is the only place it is used

Reply:
@santiagoboterokommit Yes, that is clear to me. Mixing `:ed25519` and `nil` in the same "schemas" list seems not to be very appropriate. Just allow the `nil` value in the multiclause function and that's it!